src/tracker.cpp

```diff
@@ -96,6 +98,31 @@ void Tracker::Init()
 
 void Tracker::MainLoop() {vrpn_tracker_.mainloop();}
 
+builtin_interfaces::msg::Time Tracker::get_timestamp(struct timeval vrpn_timestamp) {
+  if (this->use_vrpn_timestamps_) {
+    if (this->first_ros_timestamp_.sec == 0) {
+      this->first_ros_timestamp_ = this->get_clock()->now();
+      this->first_vrpn_timestamp_ = vrpn_timestamp;
+    }
+    builtin_interfaces::msg::Time stamp;
+    int64_t sec = vrpn_timestamp.tv_sec - this->first_vrpn_timestamp_.tv_sec + this->first_ros_timestamp_.sec;
+    int64_t nanosec = (vrpn_timestamp.tv_usec - this->first_vrpn_timestamp_.tv_usec) * 1000 + this->first_ros_timestamp_.nanosec;
```

Comment:
Although it's nice to compute the offset between the ROS clock and the VRPN clock and carry the offset for future measurements, I am a bit concerned that since you mentioned the ROS clock can be unreliable at times, the two clocks could slightly diverge when running for a long time.

How about making it simple: if this flag is turned on, use the raw VRPN timestamp without converting to the ROS clock? The user can do the conversion / synchronization on their end.

Reply:
I agree this is simpler, I'll push the new version in a few minutes